app/components/ConfirmTrade.tsx

```diff
@@ -48,9 +46,9 @@ export const ConfirmTrade = ({
   tokenAmountOut,
 }: ConfirmTradeProps) => {
   const config = useConfig();
-  const { isModalOpen, openModal, closeModal } = useModalContext();
-  const [txHash, setTxHash] = useState("");
-  const [isTxLoading, setIsTxLoading] = useState(false);
+  const { isModalOpen, closeModal } = useModalContext();
+  const { writeContractShowModal } = useWeb3Context();
```

Comment:
Here we should call is something like `useTransactionContext`or just `useTransaction`.